### ISSUE TITLE
Boost data processing for data coming from the Simulink model

### ISSUE THREAD

**Issue by gabrielenava** [robotology/whole-body-controllers#13]
With [this commit](https://github.com/robotology-playground/wholeBodyControllers/commit/b82317a0e9eeccdb7bdb3ba22e81449e2b4d43b6), I added a callback in every model to a function like [this one](https://github.com/robotology-playground/wholeBodyControllers/blob/WBT3/controllers/torqueBalancingYoga/stopTorqueBalancingYoga.m). The function is called every time the `terminate` button is pressed. The purpose was to remove the local paths that are added when running a specific simulation, e.g. Yoga or Standup. The reason is that the local paths may contain functions with the same names and this may be dangerous (it is not clear which function is called). 

However, I noticed that this callback function is run only after the `to workspace` block updates the workspace. I realized that we can then use this function to plot data, save the workspace, etc, automatically and every time we run a demo on the robot. This may be useful for processing data while on the robot, or compare two different simulations and decide if the data collected are valuable.

**Comment by gabrielenava** [comment]
I will try to implement this during this week. The Idea is the following:

- run a demo;
- terminate and collect data, plot data;
- decide whether to save or discard data.

@S-Dafarra may be useful also for you for collecting data on the robot.

**Comment by gabrielenava** [comment]
I think I will implement this soon for collecting data for IROS.

**Comment by diegoferigo** [comment]
@gabrielenava In my opinion we should exploit this features and dump all the data we can gather from the robot using simulink to the workspace in a structured way. With this I mean that the initialization script could contain a flag to enable / disable data logging, and all the blocks that export data (scopes mainly) are configured in such a way to fill a class' object.

This class can be developed separately and might provide useful methods for automatically post-processing and plotting data. If needed, serializers to different file formats (csv, ...), e.g. for committing datasets, might be integrated easily.

We can have a f2f meeting for discussing this more in detail (remember: you applied _priority low_!)

**Comment by gabrielenava** [comment]
closing as a duplicate of https://github.com/robotology-playground/whole-body-controllers/issues/20
